qiskit_experiments/test/pulse_backend.py

```diff
@@ -603,6 +603,7 @@ def __init__(
             }
         )
         self._target = Target(
+            num_qubits=1,
             qubit_properties=[QubitProperties(frequency=qubit_frequency)],
             dt=self.dt,
             granularity=16,
```

Comment:
You call `Target.add_instruction` below, and `num_qubits` is supposed to be inferred from that if it wasn't given.  What kind of error message were you seeing?  This might be a bug in Qiskit 1.3, if `num_qubits` is required and it wasn't in 1.2.

Reply:
The error is `ValueError: The value of num_qubits specified does not match the length of the input qubit_properties list` which is fair because the docstring does say that the length of `qubit_properties` needs to match `num_qubits`. I don't see an alternative way to set the qubit properties, so it seems like growing the target with `add_instruction` only works if you don't use qubit properties.

Reply:
The "Raises" documentation of the `Target` says:

> `ValueError`: If both ``num_qubits`` and ``qubit_properties`` are both defined and the value of ``num_qubits`` differs from the length of ``qubit_properties``.

but you haven't specified `num_qubits`, so it shouldn't raise an error.  I'd treat this as a bug in Qiskit, especially if it worked in 1.2.  It should be easy enough for us to fix - I imagine that we've just made a slight mistake, and are defaulting `num_qubits=0` on the way down to Rust space, rather than sending `num_qubits=None`, so Rust space can tell whether it was specified or not.

Do you mind opening a bug report on Qiskit about this, or is it easier if I do it?

Reply:
I can open an issue, but what about the description of `qubit_properties` in the docstring?

             qubit_properties (list): A list of :class:`~.QubitProperties`
                objects defining the characteristics of each qubit on the
                target device. If specified the length of this list must match
                the number of qubits in the target, where the index in the list
                matches the qubit number the properties are defined for. If some
                qubits don't have properties available you can set that entry to
                ``None``

I suppose it says that it must match the number of qubits in the target but maybe if you don't specify `num_qubits` there should be no error until you `add_instruction` something that makes the number too big?

Reply:
I just looked at the code for 1.2.4 and `main`, and it's definitely a mistake in `main`, similar to how I described.  Both 1.2.4 and `main` have explicit checks that should raise the error you saw if both `num_qubits` and `qubit_properties` are given and disagree, but the Rust-space version on `main` does it by accepting `num_qubits: Option<usize>` and checking for `None`, but the Python-space class that wraps `Target` passes `num_qubits=0` by default, _not_ `num_qubits=None`, so the error message never triggers.

The whole docstring isn't clear, but the intention (from how the code was written) was clearly that `num_qubits` should be inferred from `qubit_properties` if the latter was given and the former wasn't.  It's definitely a bug in the new implementation, and a clear missing test case from the original implementation.

Reply:
I opened https://github.com/Qiskit/qiskit/issues/13393 for this. I think it should be tagged as 1.3.0 but I left that for you.